Updates, Testing & Examples

GreenPioneer opened the pull request:
Tried to leave my comments around the code.

Still need to flush out the testing but its a start

GreenPioneer committed:
Got rid of es lint , added standardjs & updated package json

GreenPioneer committed:
added docs , travis.yml & npm ignore

GreenPioneer committed:
added a example to run for examples & testing

GreenPioneer committed:
Added the layout of the test cases with the first section working just not asserting anything

GreenPioneer committed:
Updated to standard js  & made some update while trying to leave comments near my updates.

GreenPioneer committed:
Merge Contributing into .github & Update License

joshbeckman committed:
Update CONTRIBUTING.md

joshbeckman committed:
Reserve capitalized names for classes/models

joshbeckman commented on lib/throttler.js:
```diff
@@ -10,86 +11,107 @@ var defaults      = require('../defaults'),
+
+    ip = ip.split(',')[0]
+    if (ip.substr(0, 7) === '::ffff:' || ip === '::1') {
+      // DEFUALT TO LOCALHOST SO THAT IF WE DO GET A INVALID IP WE CAN RELY ON THE WE CAN STILL LIMIT THE ATTACK OR PERSON
```
👍

GreenPioneer replied:
What are you thoughts one of these NPM IP packages handling the ip for us ?

joshbeckman replied:
see https://github.com/andjosh/mongo-throttle/commit/7b52d4b07ea45616a84da721dd605b999e9fb60e (and 60ff0ee72fe1f242763d67b64e7f7561e5bdfb1d)

GreenPioneer replied:
👍

joshbeckman commented on lib/throttler.js:
```diff
@@ -10,86 +11,107 @@ var defaults      = require('../defaults'),
+  return throttleMiddleware
+
+  function errHandler (res, next, error) {
+    // DO WE WANT TO BREAK THE APP AND CAUSE A ISSUE OR DO WE WANT TO SILENTLY INTO THE NIGHT AND JUST LOG OUR ERRORS
```
Yeah, I had the same question. For myself, I would rather error out like this.

GreenPioneer replied:
That works for me , Would it be useful in the future to add a emailer  to notify admins or another way to allow the admins to make a custom error handler ? much how we can create a custom response to the limiter

joshbeckman replied:
I think adding an emailer may be a bit much, but we can either:
- Add a flag to optionally fail silently
- Add an optional error handler

joshbeckman commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,31 @@
+<!--- Provide a general summary of your changes in the Title above -->
```
I didn't know about the `.github/` configs. This is fantastic!

GreenPioneer replied:
Yea i think it really makes it easier and i hope you like the template docs i have for it

joshbeckman committed:
Update package.json

joshbeckman commented on test/mocha.test.js:
```diff
@@ -0,0 +1,70 @@
+        .expect(200, function (err, res) {
+          if (err) return done(err)
+          assert.equal(err, null)
+          console.log(res.headers['x-rate-limit-limit'])
```
I wouldn't `log` this stuff, just because some CI runners (like Atlassian Bamboo, others) expect logged output to conform to certain formats, and I've had tests fail in the past simply because of a `console.log` "counting as error output" from a test runner.

GreenPioneer replied:
Yea im fine with that , i actually left these in cause my comp was dying and i had no more time to test so i just left it  👍

joshbeckman committed:
Merge branch 'master' of github.com:GreenPioneer/mongo-throttle into test

joshbeckman reviewed (commented):
Here's a patch to actually have some tests and remove the `console.log` statements: https://gist.github.com/andjosh/89f62e46072e5a14775e9fd9ce813949

GreenPioneer commented on .travis.yml:
```diff
@@ -0,0 +1,17 @@
+language: node_js
```
Can you turn on travis for us since its through your account

joshbeckman committed:
Update tests for no configs

Now testing for:
- setting headers
- incrementing hits
- decrementing time remaining
- setting reset times
- setting default limits

GreenPioneer commented:
Hey josh 

Catch a small snag in our configs and default. im gonna fix it and add in the test case soon.

would you be able to make me a collab on the npmjs so that i can push these type of small things with out have to pester you 

Jason


Also what tool do you use for creating the docs
> On Oct 16, 2016, at 8:58 AM, Josh <notifications@github.com> wrote:
> 
> @andjosh commented on this pull request.
> 
> In lib/throttler.js <https://github.com/andjosh/mongo-throttle/pull/1>:
> 
> >      var ip = request.headers['x-forwarded-for'] ||
> -        request.connection.remoteAddress ||
> -        request.socket.remoteAddress ||
> -        request.connection.socket.remoteAddress;
> +      request.connection.remoteAddress ||
> +      request.socket.remoteAddress ||
> +      request.connection.socket.remoteAddress || ''
> +
> +    ip = ip.split(',')[0]
> +    if (ip.substr(0, 7) === '::ffff:' || ip === '::1') {
> +      // DEFUALT TO LOCALHOST SO THAT IF WE DO GET A INVALID IP WE CAN RELY ON THE WE CAN STILL LIMIT THE ATTACK OR PERSON
> see 7b52d4b <https://github.com/andjosh/mongo-throttle/commit/7b52d4b07ea45616a84da721dd605b999e9fb60e> (and 60ff0ee <https://github.com/andjosh/mongo-throttle/commit/60ff0ee72fe1f242763d67b64e7f7561e5bdfb1d>)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/andjosh/mongo-throttle/pull/1>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AHiArokUfIBjAVPGgsk8XujSGxmzA5XAks5q0i2NgaJpZM4KXiGt>.
>

andjosh commented:
Added you onto the NPM package. What's the small snag?

What do you mean "tool for creating the docs?"

andjosh commented:
NVM, I see your issue now and fixed it in #4.

GreenPioneer commented:
https://github.com/IonicaBizau/blah <https://github.com/IonicaBizau/blah>

a tool like this , are you interested in using something that just generates our docs

i thought you had one cause you wrote the jsdoc up


> On Oct 16, 2016, at 5:19 PM, Josh <notifications@github.com> wrote:
> 
> Added you onto the NPM package. What's the small snag?
> 
> What do you mean "tool for creating the docs?"
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/andjosh/mongo-throttle/pull/1#issuecomment-254079297>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AHiArm9QKcEjmhiLRYLN_ROw5KHQuK54ks5q0qLygaJpZM4KXiGt>.
>

andjosh commented:
Ah, no. I just generate the JSDocs from a VIM plugin and then wrote the README by hand. What's some example output/input for `blah`?
